### PR TITLE
make crier a common library that can take in a reporter interface

### PR DESCRIPTION
**Pull request by krzyzacy**
- made a fake gerrit reporter as a proof of concept, will fill it up after https://github.com/kubernetes/test-infra/pull/9280, with https://github.com/kubernetes/test-infra/pull/9045
- renamed the existing crier binary to gerrit-crier, so later on we can add github-crier, pubsub-crier, as separate controllers.
- clean up binary after built component image with make rules

/area prow
/assign @cjwagner @stevekuznetsov @munnerz 

cc @yutongz this should help you to write a separete pubsub-crier

### CODE REVIEW

**Comment by BenTheElder** [comment]
generally looks fine to me, we should add a `go_image` for the gerrit-crier

**Comment by krzyzacy** [comment]
added go_image, PTAL?

File: prow/cmd/gerrit-crier/main.go [hunk 1]
```diff
@@ -133,7 +134,9 @@ func main() {
 			&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
 		))
 
-	controller := crier.NewController(client, queue, prowjobInformerFactory.Prow().V1().ProwJobs())
+	gerritReporter := reporter.NewReporter()
+
+	controller := crier.NewController(client, queue, prowjobInformerFactory.Prow().V1().ProwJobs(), gerritReporter)
```
Comment (stevekuznetsov) [inline prow/cmd/gerrit-crier/main.go]: To share the caches the controllers need to be in-process -- see upstream controller-manager codebase.

**Comment by stevekuznetsov** [comment]
@krzyzacy slightly important comment re in-process

File: prow/cmd/gerrit-crier/main.go (hunk 1 above)
Comment (krzyzacy) [reply prow/cmd/gerrit-crier/main.go]: ah... didn't really try with multiple criers, will follow up this.

File: prow/crier/controller.go [hunk 2]
```diff
@@ -160,10 +171,14 @@ func (c *Controller) processNextItem() bool {
 		return true
 	}
 
-	// TODO(krzyzacy): call report lib here
 	if pj.Spec.Report && pj.Status.PrevReportState != pj.Status.State {
 		logrus.Infof("Will report here, pj : %v, state : %s", pj.Spec.Job, pj.Status.State)
-		pj.Status.PrevReportState = pj.Status.State
+
+		// TODO(krzyzacy): we probably should make report async as well
```
[the remark this answers is not in the document]
Comment (krzyzacy) [reply prow/crier/controller.go]: yeah I figured - maybe increase the number of worker is enough.

**Pull request merged**